engine/lib/phx/src/render/mesh.rs

```diff
@@ -31,584 +40,809 @@ pub struct Computed {
-    this._refCount = (this._refCount).wrapping_sub(1);
+#[luajit_ffi_gen::luajit_ffi]
+impl Mesh {
+    #[bind(name = "Create")]
+    pub fn new() -> Mesh {
+        Mesh {
+            shared: Rf::new(MeshShared {
+                vbo: 0,
+                ibo: 0,
+                vao: 0,
+                version: 1,
+                version_buffers: 0,
+                versionInfo: 0,
+                info: Computed {
+                    bound: Box3::default(),
+                    radius: 0.0,
+                },
+                vertex: Vec::new(),
+                index: Vec::new(),
+            }),
+        }
+    }
 
-    if this._refCount == 0 && this.vbo != 0 {
-        glcheck!(gl::DeleteVertexArrays(1, &this.vao));
-        glcheck!(gl::DeleteBuffers(1, &this.vbo));
-        glcheck!(gl::DeleteBuffers(1, &this.ibo));
+    // This simply forwards calls from Lua to the Clone trait.
+    #[bind(name = "Clone")]
+    fn clone_impl(&self) -> Mesh {
+        self.clone()
     }
-}
 
-#[no_mangle]
-pub unsafe extern "C" fn Mesh_ToBytes(mesh: &mut Mesh) -> *mut Bytes {
-    let vertexCount: i32 = Mesh_GetVertexCount(mesh);
-    let indexCount: i32 = Mesh_GetIndexCount(mesh);
-    let size: u32 = 2_usize
-        .wrapping_mul(std::mem::size_of::<i32>())
-        .wrapping_add((vertexCount as usize).wrapping_mul(std::mem::size_of::<Vertex>()))
-        .wrapping_add((indexCount as usize).wrapping_mul(std::mem::size_of::<i32>()))
-        as u32;
-    let this: *mut Bytes = Bytes_Create(size);
-    Bytes_WriteI32(&mut *this, vertexCount);
-    Bytes_WriteI32(&mut *this, indexCount);
-    Bytes_Write(
-        &mut *this,
-        (*mesh).vertex.as_ptr() as *const _,
-        (vertexCount as usize).wrapping_mul(std::mem::size_of::<Vertex>()) as u32,
-    );
-    Bytes_Write(
-        &mut *this,
-        (*mesh).index.as_ptr() as *const _,
-        (indexCount as usize).wrapping_mul(std::mem::size_of::<i32>()) as u32,
-    );
-    this
-}
+    pub fn load(name: &str) -> Mesh {
+        Self::from_bytes(&mut Resource::load_bytes(ResourceType::Mesh, name))
+    }
 
-#[no_mangle]
-pub unsafe extern "C" fn Mesh_FromBytes(buf: &mut Bytes) -> Box<Mesh> {
-    let mut this = Mesh_Create();
-    let vertexCount: i32 = Bytes_ReadI32(buf);
-    let indexCount: i32 = Bytes_ReadI32(buf);
-
-    Mesh_ReserveVertexData(this.as_mut(), vertexCount);
-    Mesh_ReserveIndexData(this.as_mut(), indexCount);
-
-    (*this)
-        .vertex
-        .resize(vertexCount as usize, Vertex::default());
-    (*this).index.resize(indexCount as usize, 0);
-    Bytes_Read(
-        buf,
-        (*this).vertex.as_mut_ptr() as *mut _,
-        (vertexCount as usize).wrapping_mul(std::mem::size_of::<Vertex>()) as u32,
-    );
-    Bytes_Read(
-        buf,
-        (*this).index.as_mut_ptr() as *mut _,
-        (indexCount as usize).wrapping_mul(std::mem::size_of::<i32>()) as u32,
-    );
-
-    this
-}
+    #[bind(name = "ToBytes")]
```

Comment:
Not needed. Will be automatically done by macro.

Reply:
Right, I think this function was named differently whilst I was working on this and I forgot to remove the `bind` here once I sorted the naming.